src/objective.jl

```diff
@@ -130,36 +128,30 @@ function MOI.get(model::LinQuadOptimizer, ::MOI.ObjectiveFunction{Linear})
     return Linear(terms, get_constant_objective(model))
 end
 
-function MOI.canget(model::LinQuadOptimizer, ::MOI.ObjectiveFunction{Quad})
-    return model.obj_type == QuadraticObjective
-end
-
 function MOI.get(model::LinQuadOptimizer, ::MOI.ObjectiveFunction{Quad})
-    if model.obj_type != QuadraticObjective
-        error("Cannot get ScalarQuadratic objective function as the model has " *
-              " a $(model.obj_type) objective function.")
-    end
     variable_coefficients = zeros(length(model.variable_references))
     get_linear_objective!(model, variable_coefficients)
     affine_terms = map(
         (variable, coefficient) -> MOI.ScalarAffineTerm{Float64}(coefficient, variable),
         model.variable_references,
         variable_coefficients
     )
-    Q = get_quadratic_terms_objective(model)
-    rows = rowvals(Q)
-    coefficients = nonzeros(Q)
     quadratic_terms = MOI.ScalarQuadraticTerm{Float64}[]
-    sizehint!(quadratic_terms, length(coefficients))
-    for (column, variable) in enumerate(model.variable_references)
-        for j in nzrange(Q, column)
-            row = rows[j]
-            push!(quadratic_terms,
-                  MOI.ScalarQuadraticTerm{Float64}(
-                      coefficients[j],
-                      model.variable_references[row],
-                      variable)
-            )
```

Comment:
Why there is no more objective type?

Reply:
What do you mean?
Notice that I modified the behavior of this function according to the requirement of `MOI.canget(model::LinQuadOptimizer, ::MOI.ObjectiveFunction{Quad})`: an optimizer must return a proper quadratic function even when all the quadratic terms are zero (i.e. the function is actually linear or a single variable).

Reply:
ok!